core/src/main/java/nu/yona/server/subscriptions/entities/UserAnonymized.java

```diff
@@ -66,8 +67,13 @@ public MessageDestination getAnonymousDestination()
 
 	public Set<MessageDestination> getBuddyDestinations()
 	{
-		return buddiesAnonymized.stream().filter(ba -> ba.getSendingStatus() == Status.ACCEPTED).map(ba -> ba.getUserAnonymized())
-				.map(ua -> ua.getAnonymousDestination()).collect(Collectors.toSet());
+		return mapAvailableDestinations(buddiesAnonymized.stream()).collect(Collectors.toSet());
+	}
+
+	public static Stream<MessageDestination> mapAvailableDestinations(Stream<BuddyAnonymized> buddiesAnonymized2)
```

Comment:
The proposed solution is hard because these anonymized buddy DTOs need to expire when the buddy status changes (the `BuddyService` needs to expire them). Maybe the solution is to add a cache for anonymized buddy DTOs maintained by the `BuddyService` to accomplish this.